components/script/dom/document.rs

```diff
@@ -516,6 +516,25 @@ impl<'a> DocumentMethods for JSRef<'a, Document> {
     fn URL(self) -> DOMString {
         self.url().serialize()
     }
+    
+    // https://html.spec.whatwg.org/multipage/interaction.html#dom-document-activeelement
+    fn GetActiveElement(self) -> Option<Temporary<Element>> {
+        // TODO: Step 2.
+
+        match self.get_focused_element() {
+            Some(element) => Some(element),     // Step 3. and 4.
+            None => match self.GetBody() {      // Step 5.
+                Some(body) => Some(ElementCast::from_temporary(body)),
+                None => {
+                    let node = NodeCast::from_ref(self);
+                    match node.GetFirstChild() {
+                        Some(_) => ElementCast::to_ref(node).and_then(ElementHelpers::get_root_element),
```

Comment:
The first line of that link is "The term root element, when not referring to a Document object's root element,", and in this case the original text makes me believe this is what we want.